models/db/list.go

```diff
@@ -133,27 +133,28 @@ type FindOptionsOrder interface {
 
 // Find represents a common find function which accept an options interface
 func Find[T any](ctx context.Context, opts FindOptions) ([]*T, error) {
-	sess := GetEngine(ctx)
+	sess := GetEngine(ctx).Where(opts.ToConds())
```

Comment:
`JoinFunc` is a `func(sess Engine)` which can be called with `xorm.Engine` and `xorm.Session`. Inside the join func `sess.Join` is used. Now the problem is that `Session.Join` adds the `JOIN` to the statement while `Engine.Join` creates a new `Session` which gets discarded afterwards. So calling the `JoinFunc` with `xorm.Engine` does "nothing" while calling with `xorm.Session` does what it should.

`GetEngine(ctx)` returns a `xorm.Engine` and `Where(...)` creates the needed `Session`.